subsys/bluetooth/host/gatt.c

```diff
@@ -1671,6 +1685,31 @@ static u8_t notify_cb(const struct bt_gatt_attr *attr, void *user_data)
 
 	ccc = attr->user_data;
 
+	/* Save Service Changed data if peer is not connected */
+	if (IS_ENABLED(CONFIG_BT_GATT_SERVICE_CHANGED) && ccc == &sc_ccc) {
+		for (i = 0; i < ARRAY_SIZE(sc_cfg); i++) {
+			struct gatt_sc_cfg *cfg = &sc_cfg[i];
+			struct bt_conn *conn;
+
+			if (!bt_addr_le_cmp(&cfg->peer, BT_ADDR_LE_ANY)) {
+				continue;
+			}
+
+			conn = bt_conn_lookup_addr_le(cfg->id,
+						      &cfg->peer);
+			if (!conn) {
+				struct sc_data *sc;
```

Comment:
It is done 2 lines above (see if (!conn)).

In this callback, we check if the device we want to send the indication to is connected. If it is, we send the indication, otherwise we save the indication data in the SC setting so we can remember to send the indication the next time the device connects.

Reply:
Yes, finding a connection object doesn't automatically mean that you're connected. There may be a left-over reference but the state is actually disconnected.

Reply:
There's a `bt_conn_lookup_state_le()` that's probably most convenient here

Reply:
Updated to use bt_conn_lookup_state_le() in latest push, to insure we save SC data in the appropriate case.